### PR TITLE
Add io.github.mrmyhuang.cbetar2 flatpak.

### PR DESCRIPTION
**Pull request by MrMYHuang**
### Please confirm your submission meets all the criteria

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [ ] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
   * This app supports exporting/importing app settings to/from file. Thus, it needs `--filesystem=home`.
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [X] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by hfiguiere** [comment]
>     This app supports exporting/importing app settings to/from file. Thus, it needs --filesystem=host.

Why host?

File: flathub.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (hfiguiere) [inline flathub.json:1]: you don't need that file

File: io.github.mrmyhuang.cbetar2.desktop [hunk 2]
```diff
@@ -0,0 +1,8 @@
+[Desktop Entry]
+Name=電子佛典
+GenericName=CBETA電子佛典2
+Exec=/app/bin/run.sh
```
Comment (hfiguiere) [inline io.github.mrmyhuang.cbetar2.desktop:4]:
```suggestion
Exec=run.sh
```

File: io.github.mrmyhuang.cbetar2.yml [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+          - zypak-wrapper.sh /app/main/cbetar2 "$@"
+
+      - type: file
+        path: io.github.mrmyhuang.cbetar2.metainfo.xml
+
+      - type: file
+        path: io.github.mrmyhuang.cbetar2.desktop
+
+      - type: file
+        path: io.github.mrmyhuang.cbetar2.png
```
Comment (hfiguiere) [inline io.github.mrmyhuang.cbetar2.yml:55]: the repository has this file so you shouldn't need to add it to the repository

File: io.github.mrmyhuang.cbetar2.desktop [hunk 4]
```diff
@@ -0,0 +1,8 @@
+[Desktop Entry]
```
Comment (hfiguiere) [inline io.github.mrmyhuang.cbetar2.desktop:1]: this file should be submitted upstream to be handle there. A Linux desktop app that doesn't have a .desktop, it's a bug.

File: io.github.mrmyhuang.cbetar2.metainfo.xml [hunk 5]
```diff
@@ -0,0 +1,30 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (hfiguiere) [inline io.github.mrmyhuang.cbetar2.metainfo.xml:1]: this file should be also submitted upstream to be managed there. It's not flathub specific

File: io.github.mrmyhuang.cbetar2.metainfo.xml [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+<component type="desktop-application">
+  <id>io.github.mrmyhuang.cbetar2</id>
+  <name>CBETA電子佛典閱讀器2(非官方)</name>
+  <summary>此電子佛典閱讀器 app 使用 CBETA API 存取佛經資料。</summary>
+  <metadata_license>MIT</metadata_license>
+  <project_license>MIT</project_license>
+  <description>
+    <p>
+      特色：搜尋目錄、全文檢索、書籤功能、網址分享、離線瀏覽、語音播放、佛學詞典、佈景主題切換、經文分頁、字型調整、楷書字型、直排文字、列印經文/抄經本、連線/離線DB支援、跨平台、無廣告、開放原始碼。
```
Comment (hfiguiere) [inline io.github.mrmyhuang.cbetar2.metainfo.xml:10]: a description in english would be appreciated.

File: io.github.mrmyhuang.cbetar2.yml [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+  - --share=ipc
+  - --socket=x11
+  - --socket=pulseaudio
+  - --share=network
+  - --device=dri
+  - --filesystem=host
+build-options:
+  append-path: /usr/lib/sdk/node14/bin
+  cflags: -O2 -g
+  cxxflags: -O2 -g
```
Comment (hfiguiere) [inline io.github.mrmyhuang.cbetar2.yml:21]:
```suggestion
```

you don't need these.

File: io.github.mrmyhuang.cbetar2.yml [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+  - org.freedesktop.Sdk.Extension.node14
+command: run.sh
+separate-locales: false
+finish-args:
+  - --share=ipc
+  - --socket=x11
+  - --socket=pulseaudio
+  - --share=network
+  - --device=dri
+  - --filesystem=host
```
Comment (hfiguiere) [inline io.github.mrmyhuang.cbetar2.yml:17]: this is a very broad permission.

**Review by hfiguiere — CHANGES_REQUESTED** [review]

**Comment by hfiguiere** [comment]
bot, build io.github.mrmyhuang.cbetar2

File: flathub.json (hunk 1 above)
Comment (MrMYHuang) [reply flathub.json:1]: fix: delete flathub.json.

File: io.github.mrmyhuang.cbetar2.desktop (hunk 2 above)
Comment (MrMYHuang) [reply io.github.mrmyhuang.cbetar2.desktop:4]: fix

File: io.github.mrmyhuang.cbetar2.yml (hunk 7 above)
Comment (MrMYHuang) [reply io.github.mrmyhuang.cbetar2.yml:21]: fix

File: io.github.mrmyhuang.cbetar2.metainfo.xml (hunk 5 above)
Comment (MrMYHuang) [reply io.github.mrmyhuang.cbetar2.metainfo.xml:1]: fix

File: io.github.mrmyhuang.cbetar2.desktop (hunk 4 above)
Comment (MrMYHuang) [reply io.github.mrmyhuang.cbetar2.desktop:1]: fix

File: io.github.mrmyhuang.cbetar2.yml (hunk 3 above)
Comment (MrMYHuang) [reply io.github.mrmyhuang.cbetar2.yml:55]: fix

File: io.github.mrmyhuang.cbetar2.yml (hunk 8 above)
Comment (MrMYHuang) [reply io.github.mrmyhuang.cbetar2.yml]: - --socket=pulseaudio: the app uses text-to-speech function.
  - --share=network: this app needs Internet to access the Buddhism text database.
  - --device=dri: deleted.
  - --filesystem=host: change host to home

Comment (hfiguiere) [reply io.github.mrmyhuang.cbetar2.yml]: actualy for electron DRI is likely required. If you get messages about about GL or libdrm, it's because of that.

**Comment by hfiguiere** [comment]
bot, build io.github.mrmyhuang.cbetar2

File: io.github.mrmyhuang.cbetar2.yml (hunk 8 above)
Comment (MrMYHuang) [reply io.github.mrmyhuang.cbetar2.yml]: I try to build and run my app with and without dri, but the terminal messages (including errors) seems to be no different... However, my app works well in both cases. Thus, I remove the permission. Thank you.

**Comment by MrMYHuang** [comment]
> Build [80886 successful](https://flathub.org/builds/#/builders/32/builds/80886) To test this build, install it from the testing repository:
> 
> ```
> flatpak install --user https://dl.flathub.org/build-repo/78774/io.github.mrmyhuang.cbetar2.flatpakref
> ```

By my test, both x86_64 and aarch64 builds work well. Thanks.

**Comment by barthalion** [comment]
FYI @mwleeds I'm observing some regressions with applications built using libappstream-based flatpak and then installed wth regular flatpak 1.12.x. Applications which use `/app/share/metainfo` seem to produce the following warning during installation:

```
[bpiotrowski@risuku ~]$ flatpak install --user https://dl.flathub.org/build-repo/78784/com.github.ahrm.sioyek.flatpakref

com.github.ahrm.sioyek permissions:
    ipc     x11     dri    shm    file access [1]

    [1] /media, /run/media, /tmp, xdg-documents, xdg-download, xdg-music


        ID                              Branch          Op          Remote                Download
 1. [✓] com.github.ahrm.sioyek          test            i           sioyek-origin         103,8 MB / 104,6 MB

Warning: Not exporting file com.github.ahrm.sioyek.appdata.xml of unsupported type
```

I don't see anything functionally wrong with this manifest so I think the problem is in Flatpak itself, but maybe I missed something?. The same happens with https://github.com/flathub/flathub/pull/2897.

**Comment by barthalion** [comment]
@MrMYHuang Can you please change the `<summary>` tag to use English too?

**Comment by MrMYHuang** [comment]
bot, build io.github.mrmyhuang.cbetar2

**Comment by MrMYHuang** [comment]
> @MrMYHuang Can you please change the `<summary>` tag to use English too?

Yes. Please see:
https://github.com/MrMYHuang/cbetar2/commit/b25f4e77f50a4289d6ba9e2b51ce9b3e8fbe79db
Thank you.

**Comment by MrMYHuang** [comment]
Sorry, I made a mistake in commit https://github.com/flathub/flathub/pull/2901/commits/44210b6169005a19e53bfda24507c31e44576e22
Please help me stopping the build https://flathub.org/builds/#/builders/32/builds/80936
Thanks.

The updated commit is https://github.com/flathub/flathub/pull/2901/commits/79dcfa84b0c4169a2e4589f10a806e60d10ec121

**Comment by MrMYHuang** [comment]
bot, build io.github.mrmyhuang.cbetar2

**Comment by MrMYHuang** [comment]
bot, build io.github.mrmyhuang.cbetar2

**Comment by MrMYHuang** [comment]
> Build [80954 successful](https://flathub.org/builds/#/builders/32/builds/80954) To test this build, install it from the testing repository:
> 
> ```
> flatpak install --user https://dl.flathub.org/build-repo/78839/io.github.mrmyhuang.cbetar2.flatpakref
> ```

I tested the both x86_64 and aarch64 builds. They work well.

**Pull request closed without merging**